crates/wasmtime/src/func.rs

```diff
@@ -273,11 +273,15 @@ macro_rules! generate_wrap_async_func {
         {
             assert!(store.as_context().async_support(), concat!("cannot use `wrap", $num, "_async` without enabling async support on the config"));
             Func::wrap(store, move |mut caller: Caller<'_, T>, $($args: $args),*| {
-                let async_cx = caller.store.as_context_mut().0.async_cx();
-                let mut future = Pin::from(func(caller, $($args),*));
-                match unsafe { async_cx.block_on(future.as_mut()) } {
-                    Ok(ret) => ret.into_fallible(),
-                    Err(e) => R::fallible_from_trap(e),
+                match caller.store.as_context_mut().0.async_cx() {
+                    None => R::fallible_from_trap(Trap::from(anyhow::format_err!("Attempt to start async function on dying fiber"))),
```

Comment:
For this I think it's ok to do `.expect()` because any trap raised from cancellation of a fiber should not be catch-able by wasm and wasm should never be able to initiate a call of an async import.

Reply:
Cool. I've pulled these checks and replaced them with `expect()` (mostly; a couple `unwrap`s) in a775e97.